src/types/Processor.ts

```diff
@@ -25,9 +25,40 @@ export interface EventFilter {
   value: string;
 }
 
-export interface BaseSource {
+export interface ProcessorRequestData {
+  name: string;
   type: string;
-  parameters: {
+  action: Action;
+  transformationTemplate?: string;
+  filters?: EventFilter[];
+  source?: Source;
+}
+
+export interface ProcessorValidation {
+  isValid: boolean | undefined;
+  errors: {
     [key: string]: string;
   };
 }
+
+export interface ConfigType {
+  value: string;
```

Comment:
I found it a little confusing having this as `value` whereas it is checked against (type) `name`'s.

Just an opinion; no need to change it.